### ISSUE TITLE
[Bug]: dn/cn crash by runtime error: invalid memory address or nil pointer dereference when run tpcc 100-100 for about 15 minutes

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#9862]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):4f6cf9a
- Hardware parameters:
  3*CN : 16C 64G
  1*DN : 16C 64G
  3*LOG: 16C 64G
- OS type:
- Others:
  DISKCACHE: 100G
  MEMCACHE: 16G
```

### Actual Behavior

when running tpcc 100-100 on EKS with 3*CNs for about 15 minutes, the CN and DN crashed.
<img width="596" alt="image" src="https://github.com/matrixorigin/matrixone/assets/47846308/adeb271a-67ad-41f9-a1e7-f61523eb5f08">

the log:
[cn-crash.log](https://github.com/matrixorigin/matrixone/files/11654152/cn-crash.log)
[dn-crash.log](https://github.com/matrixorigin/matrixone/files/11654153/dn-crash.log)

### Expected Behavior

CN and DN should work normally

### Steps to Reproduce

```Markdown
1. run a mo cluster on EKS with 3*CN and using the same resource
2. run tpcc 100 warehouse and 100 terminals for about 20 minutes
```

**Comment by cnutshell** [comment]
```go
409             r.blks[0].deletes = append(r.blks[0].deletes, int64(offset))
```
It is possible that ` r.blks[0]` is nil.

<img width="1439" alt="image" src="https://github.com/matrixorigin/matrixone/assets/20291742/10e87af0-f11f-4b93-a763-d5d0d8b18cab">

**Comment by cnutshell** [comment]
Now, logtail server would release `logtail.MessageSegment` on morpc error.
Maybe logtail server shouldn't release them, just leave the task to morpc.

```go
156         if err := s.cs.Write(ctx, seg); err != nil {
157             s.segments.Release(seg)
158             return err
159         }
```
<img width="1439" alt="image" src="https://github.com/matrixorigin/matrixone/assets/20291742/ee7235c4-757a-4863-91b7-570e6efe74a4">

**Comment by cnutshell** [comment]
<img width="1439" alt="image" src="https://github.com/matrixorigin/matrixone/assets/20291742/10e87af0-f11f-4b93-a763-d5d0d8b18cab">

For commit `4f6cf9a`,  line 409 is written as below:
```go
409             r.blks[0].deletes = append(r.blks[0].deletes, int64(offset))
```

- if `r.blks` was nil, then `r.blks[0]` would panic, but the panic should be `panic: runtime error: index out of range;
- type of `r.blks[0]` is a struct, not a pointer to that struct, so `r.blks[0].deletes` shouldn't panic;
- if `r.blks[0].deletes` was nil, `r.blks[0].deletes = append(r.blks[0].deletes, int64(offset))` shouldn't panic also;
- then, the only cause left for `panic: runtime error: invalid memory address or nil pointer dereference` should be that `r` was nil. But that's impossible according to `pkg/vm/engine/disttae.(*blockMergeReader).Read`.

That's impossible to figure out the root cause now.

**Comment by cnutshell** [comment]
@aressu1985 Please check this issue fixed or not.

**Comment by cnutshell** [comment]
@triump2020 PTAL

**Comment by triump2020** [comment]
PR is on the way.

**Comment by triump2020** [comment]
progress on

**Comment by triump2020** [comment]
I will push the PR at 6.14

**Comment by triump2020** [comment]
@aressu1985  pls test

**Comment by aressu1985** [comment]
fixed:nightly-d4d74e7
